### ISSUE TITLE
Virtual generated KeyCode_Up/KeyCode_Down not working properly in listers after user start typing.

### ISSUE THREAD

**Issue by erpeo93** [4coder-editor/4coder#266]
Version: 4.1.6

I've remapped alt + J/L/I/K to act as arrows:

```
CUSTOM_COMMAND_SIG(custom_move_up)
{
    Input_Event event  = {};
    event.virtual_event = true;
    event.kind = InputEventKind_KeyStroke;
    event.key.code = KeyCode_Up;
    enqueue_virtual_event(app, &event);
}

Bind(custom_move_up,  KeyCode_I, KeyCode_Alt);
```

When I am running a lister in a panel, they work perfectly _until_ I start typing text and the result are filtered. At that point if I send the custom_move_up command, the result are "unfiltered" with the lister_call_refresh_handler in this piece of code: (4coder_lister_base.cpp).
It seems that for some reason it's the "i" keyboard event that allows the code to get here.

```
        if (!handled){
            Mapping *mapping = lister->mapping;
            Command_Map *map = lister->map;
            Fallback_Dispatch_Result disp_result =
                fallback_command_dispatch(app, mapping, map, &in);
            if (disp_result.code == FallbackDispatch_DelayedUICall){
                call_after_ctx_shutdown(app, view, disp_result.func);
                break;
            }
            if (disp_result.code == FallbackDispatch_Unhandled){
                leave_current_input_unhandled(app);
            }
            else{
                lister_call_refresh_handler(app, lister);
            }
        }

```
Basically I can navigate up and down in lister with virtual generated keycode_up/down events only if the result remain unfiltered: at that point.

NOTE: I've binded the custom move commands in global, file, and code mappings.

Step to repro: just execute the custom_move_up command I posted while you're on a lister and you have started filtering the result. (for example the open file lister) 

OS: Win10
